lib/CustomElementRegistry.js

```diff
@@ -0,0 +1,155 @@
+"use strict";
+var xml = require('./xmlnames');
+
+var forbiddenNames = [
+  'annotation-xml',
+  'color-profile',
+  'font-face',
+  'font-face-src',
+  'font-face-uri',
+  'font-face-format',
+  'font-face-name',
+  'missing-glyph'];
+
+var forbiddenExtendsNames = [
+  'bgsound',
+  'blink',
+  'isindex',
+  'multicol',
+  'nextid',
+  'spacer',
+  'elementnametobeunknownelement'
+];
+
+function isInvalidCustomElementName(name) {
+  return !name || !xml.isValidQName(name) || name.indexOf('-') < 0 || forbiddenNames.indexOf(name) > -1 || name.toLowerCase() !== name;
+}
+
+function CustomElementRegistry(registry, document) {
```

Comment:
I'm also new :) but as I see domino should work with node 0.8 so in code the class syntax is not used, unfortunately, I had to use class syntax in one place and I do not know how to avoid it :(